pigosat.go

```diff
@@ -312,3 +329,80 @@ func (p *Pigosat) BlockSolution(solution Solution) error {
 	p.blocksol(solution)
 	return nil
 }
+
+// Write the clauses that were used in deriving the empty clause to a file
+// in DIMACS format.
+//
+// Requires that Pigosat was created with EnableTrace == true.
+func (p *Pigosat) WriteClausalCore(f io.Writer) error {
+	if p.Res() != Unsatisfiable {
+		return errors.New("expected to be in Unsatisfiable state")
+	}
+	defer p.ready(true)()
```

Comment:
Lock needs to be obtained before calling `p.Res()` to avoid race condition between `p.Solve` and this line of code.